pkgs/applications/networking/iroh/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, cmake
+, protobuf
+, llvmPackages
+, rocksdb
+, Security
+, DiskArbitration
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "iroh";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "n0-computer";
+    repo = "iroh";
+    rev = "v${version}";
+    sha256 = "sha256-hw9OTV+IE4TE3Ubtpw6HiUmapBYAKN33xgVA1siCgqM=";
+  };
+
+  cargoPatches = [
```

Comment:
If we go that route, I'd probably just ship the Cargo.lock in nixpkgs directly, and refer to it via https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file